feat(MTMR): document the new MTMR conversion tool

danila-m opened the pull request.

danila-m committed:
feat(MTMR): Create the doc page for MTMR tool

danila-m committed:
Put the doc page in the right place

danila-m committed:
fix typo

educhastenier reviewed (commented):
* This page should explain in detail what needs to be done: installation of multiple runtimes, configuration of each database, etc...
* This PR should be done on 2021.1, which is the latest supported

educhastenier committed:
feat(MTMR conversion tool): add general idea schema

to better explain the idea of the tool

Sources of the image can be found at https://docs.google.com/drawings/d/1JbhaYP7u5exUmJjd0bsM8tPSpYJEFAdCdoM69g4J54Q
to modify the image if necessary

Shmayro committed:
add human friendly durations to mysql highlights

educhastenier commented on modules/data/pages/contracts-and-contexts.adoc:
```diff
@@ -123,7 +123,7 @@ In `Edit` mode, additional `persistenceId_string` input are generated to ensure
 The following example describes the logic:
 
-image::images/formGenerationReadOnly.svg[Read only example]
+image:images/formGenerationReadOnly.svg[Read only example]
```
Fixing this because the image did not show anymore, since conversion from MD to Asciidoc 😞

vbotnari12 committed:
Update mtmr-tool.adoc

Some adjustments.

vbotnari12 reviewed (changes requested):
Preview link is not available. Something may be wrong with the navigation.

danila-m commented:
> Preview link is not available. Something may be wrong with the navigation.

You can find it here --> http://bonitasoft-bonita-doc-build_preview-pr-2080.surge.sh/bonita/2022.2/version-update/mtmr-tool

vbotnari12 committed:
Update mtmr-tool.adoc

danila-m commented on modules/version-update/pages/mtmr-tool.adoc:
```diff
@@ -5,13 +5,23 @@
-The tool will delete everything not related to the tenant specified in the parameter from the given database, this means you will need to dump your current database and mount one new database dump per tenant in your environment. The tool will have to be executed for every database dump.
+
+
+Following the commercial depreciation of multi-tenancy use by customer willing to have separate BDMs, the technical depreciation will be effective with Bonita 2023.1. The technical solution for customers willing to have separate BDMs per applications/processes proposed by Bonitasoft will be multi-runtimes. 
```
"for customers willing to have separate BDMs per applications/processes" --> I am not certain that is correct.
If it means BDM as in the BDM database, it's just wrong/weird.
If it means something else in this context, we need to use another term or explicit the abbreviation, because BDM in the documentation always means the BDM database.

vbotnari12 replied:
Changed it with Bonita engine database. Adrien's suggestion :)

danila-m commented on modules/version-update/pages/mtmr-tool.adoc:
```diff
@@ -0,0 +1,523 @@
+Following the commercial depreciation of multi-tenancy use by customer willing to have separate BDMs, the technical depreciation will be effective with Bonita 2023.1. The technical solution for customers willing to have separate BDMs per applications/processes proposed by Bonitasoft will be multi-runtimes. 
+
+
+In order to prepare the technical conversion from multi-tenancy to multi-runtime, Bonitasoft developped a dedicated tool, that will have to be run as many times as the number of the tenants to be kept. 
```
```suggestion
In order to prepare the technical conversion from multi-tenancy to multi-runtime, Bonitasoft developped a dedicated tool, that will have to be run once for every tenant to be kept. 
```

vbotnari12 committed:
Update modules/version-update/pages/mtmr-tool.adoc

Co-authored-by: danila_mazour <danila.mazour@bonitasoft.com>

akantcheff commented on modules/version-update/pages/mtmr-tool.adoc:
```diff
@@ -0,0 +1,523 @@
+In order to prepare the technical conversion from multi-tenancy to multi-runtime, Bonitasoft developped a dedicated tool, that will have to be run once for every tenant to be kept. 
+
+
+The tool will delete everything not related to the tenant specified in the parameter from the given database, this means you will need to dump your current database and mount one new database dump per tenant in your environment. 
```
Some confusions for me:
- To prevent some misunderstanding, maybe we should explicitly say that the customer will need to set up a new database instance per tenant. And then load the dump onto it.
- By "environment" I might understand a runtime server and a database server. But we can use the same runtime server to run multiple runtime instances.

vbotnari12 replied:
Suggestion accepted :)

akantcheff commented on modules/version-update/pages/mtmr-tool.adoc:
```diff
@@ -0,0 +1,523 @@
+=== Pre-requisites
+* The tool requires Java 11+ JRE
+* A database of a Bonita Multi-tenant Platform. +
+The aim is to restore a dump of your production database where you want one of your new Bonita Runtimes to run in the future, and then to run the tool on it.
```
Suggestion: move this sentence in the overview and add a new prerequisites line saying something like "A dump, from actual production database, loaded to the new database"

vbotnari12 replied:
Suggestion accepted :)

vbotnari12 committed:
Update mtmr-tool.adoc

vbotnari12 committed:
Update mtmr-tool.adoc

vbotnari12 committed:
Update mtmr-tool.adoc

vbotnari12 committed:
Update mtmr-tool.adoc

vbotnari12 committed:
Update mtmr-tool.adoc

vbotnari12 committed:
Update mtmr-tool.adoc

vbotnari12 committed:
Delete mtmr-general-idea-schema.svg

vbotnari12 committed:
Add files via upload

vbotnari12 committed:
Update mtmr-tool.adoc

vbotnari12 committed:
Update mtmr-tool.adoc
